internal/crypto/cert.go

```diff
@@ -59,3 +65,48 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// CompleteCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func CompleteCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	// if certbundle is not set, return the original certs
+	if !ok {
+		return originalCerts, nil
+	}
+	certBundlePath = strings.Trim(certBundlePath, " ")
```

Comment:
We should check if the file name is cross-platform compatible, something like:
```
// IsValidFileName checks if a file name is cross-platform compatible
func IsValidFileName(fileName string) bool {
	return regexp.MustCompile(`^[a-zA-Z0-9_.-]+$`).MatchString(fileName)
}

```

Reply:
Added.